### ISSUE TITLE
Best way to adjust or lightweight metadata in 'media_sync' table

### ISSUE THREAD

**Issue by woozu-shin** [meeb/tubesync#470]
### Version
- 0.13.3

### Abstract

- The more videos there are, the larger the media_sync table becomes.
- The reason for the bloat is that each metadata json is hundreds of KB.
- It need a way to solve this.

### Detail

- I am syncing a source with about 1500 videos.
- Each sync_media record has a data size of approximately 500KB.
- The reason is because of the bloat of metadata, and a way to reduce this is needed.
- The sqlite file reached 700MB, and CPU usage and memory usage were observed to be high due to the overhead caused by - reading/dumping the huge metadata json.

### AS-IS
![tubesync_1_huge_db](https://github.com/meeb/tubesync/assets/93126826/da4364f8-b17e-4680-8e8c-647d58af8546)

### TO-BE
- There are many unverified and unworked Mock parts to create P/R.
- I am attaching a sample of my work locally.
![tubesync_2_redueced](https://github.com/meeb/tubesync/assets/93126826/74492e6e-a4a9-4de8-a4bf-7d235b1d625d)

### Sample Source 

- models.py
```python
class Source(models.Model):
    ...
    LIGHTWEIGHT_METADATA_TYPE_RAW = 'RAW'
    LIGHTWEIGHT_METADATA_TYPE_FEATHER = 'FEATHER'
    LIGHTWEIGHT_METADATA_TYPES = (LIGHTWEIGHT_METADATA_TYPE_RAW, LIGHTWEIGHT_METADATA_TYPE_FEATHER)
    LIGHTWEIGHT_METADATA_TYPE_CHOICES = (
        (LIGHTWEIGHT_METADATA_TYPE_RAW, _("(LARGE) Save raw metadata")),
        (LIGHTWEIGHT_METADATA_TYPE_FEATHER, _("(TINY) if the capacity is large, Treeshake it event if it is in use")),
    )

    lightweight_metadata = models.CharField(
        _('lightweight metadata'),
        max_length=20,
        default=LIGHTWEIGHT_METADATA_TYPE_RAW,
        choices=LIGHTWEIGHT_METADATA_TYPE_CHOICES,
        help_text=_('Lightweight metadata')
    )
```

- tasks.py
```python
        if source.lightweight_metadata == Source.LIGHTWEIGHT_METADATA_TYPE_FEATHER:
            del media.metadata["formats"]
            del media.metadata["thumbnails"]
            del media.metadata["automatic_captions"]
            del media.metadata["requested_formats"]
            del media.metadata["heatmap"]
```

### Sample View

- Add/Edit source
![image](https://github.com/meeb/tubesync/assets/93126826/35a3853d-76d4-4850-9c85-42f2f49fa627)

- Media item view (one of the media details)
![image](https://github.com/meeb/tubesync/assets/93126826/974f0c09-fe79-4c18-9d12-d6ad26ec9399)

**Comment by meeb** [comment]
If you delete formats and thumbnails from the metadata then thumbnails can't be downloaded and downloading media won't work as the media format can't be evaluated. This occurs on model save at the moment to determine if an item can be downloaded when there's a match for the requested format. While you may want to ignore the thumbnails, currently the formats (which get refreshed as and when the metadata is updated) are required.

Over the years I've had a good look at the large metadata myself, probably the most sensible may be to move it out of the database and store them as msgpack'd blobs on disk in the config dir or similar. There isn't much you can truncate from the metadata without losing functionality. You can save 5-10% but that never really seemed that worth it.

**Comment by woozu-shin** [comment]
현재로써는 이 부분을 수정하기 위해 많은 작업이 필요하다는 것은 인지하고 있다.
sybc_media metadata 가 
내부 저장된 데이터를 활용해서 내부 통신에서 활용할 수 있다면 동의한다.
하지만 이 정보들이 정말로 인하우스에 존재해야하는 정보인가? 라고 생각하면 나는 아니라고 생각한다.
sync_source 를 추가 한 이후
비동기 혹은 배치 처리를 위해 미리 metadata 를 구비하고 background task 가 해당 데이터를 이용해 다운로드를 진행하는 방식이다.

결국 redownload 에 필요한 경우인데
outbound 네트워크 통신을 수행하기 때문에 metadata 를 다시 받으면 되는 문제라고 생각한다.


아래 사진은

**Comment by meeb** [comment]
There isn't a sane way to reduce the metadata storage (it'll need to be in the SQLite database or stored per-media-item on disk) without significantly reducing generally used functionality so while it is massive and likely needs some work I'll close this as wont-fix for now. Thanks for the issue and analysis!

**Comment by tcely** [comment]
This approach was implemented in #612.
